### PR TITLE
Tolerance added

### LINKED ISSUE #601: Add tolerance
tolerate x% of deviation from aggregate and dont give any penalty, here x is a governance parameter.
Set default to 2%.
Ref #592

### PR DESCRIPTION
**Pull request by SamAg19**
fixes #601

### CODE REVIEW

File: contracts/Core/AssetManager.sol [hunk 1]
```diff
@@ -6,6 +6,7 @@ import "../IDelegator.sol";
 import "./storage/AssetStorage.sol";
 import "./parameters/child/AssetManagerParams.sol";
 import "./StateManager.sol";
+import "hardhat/console.sol";
```
Comment (abhishekvispute) [inline contracts/Core/AssetManager.sol:9]: pls remove this if not needed

**Review by abhishekvispute — COMMENTED** [review]
All else LGTM

File: contracts/Core/AssetManager.sol (hunk 1 above)
Comment (SamAg19) [reply contracts/Core/AssetManager.sol]: Removed https://github.com/razor-network/contracts/pull/628/commits/8b57244bb0715620c13670fa9bb144b50ef6fe8a

**Review by abhishekvispute — DISMISSED** [review]

File: contracts/Core/RewardManager.sol [hunk 2]
```diff
@@ -119,10 +123,15 @@ contract RewardManager is Initializable, Constants, RewardManagerParams, IReward
...
             uint64 prod = age * voteValueLastEpoch;
+            // slither-disable-next-line calls-loop
+            uint16 tolerance = assetManager.getCollectionTolerance(i);
+            tolerance = tolerance <= maxTolerance ? tolerance : maxTolerance;
+            uint64 maxVoteTolerance = medianLastEpoch + ((medianLastEpoch * tolerance) / BASE_DENOMINATOR);
+            uint64 minVoteTolerance = medianLastEpoch - ((medianLastEpoch * tolerance) / BASE_DENOMINATOR);
             // if (voteWeightLastEpoch > 0) {
-            if (voteValueLastEpoch > medianLastEpoch) {
+            if (voteValueLastEpoch > maxVoteTolerance) {
                 penalty = penalty + (prod / medianLastEpoch - age);
```
Comment (SkandaBhat) [inline contracts/Core/RewardManager.sol:133]: Penalties must be smoother.

Comment (SamAg19) [reply contracts/Core/RewardManager.sol]: Done https://github.com/razor-network/contracts/pull/628/commits/3de20dd7c241a498284f6d7053828cc1bf6f2fda

**Review by abhishekvispute — APPROVED** [review]

**Review by hrishikeshio — APPROVED** [review]

**Pull request merged**